File: app/src/main/java/com/example/paetz/yacguide/database/PartnerDao.java
package com.example.paetz.yacguide.database;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;

@Dao
public interface PartnerDao {
    @Query("SELECT * FROM Partner")
    Partner[] getAll();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(Partner partner);

    @Query("SELECT Id FROM Partner WHERE Name = :name")
    int getId(String name);

    @Query("SELECT * FROM Partner WHERE Id = :id")
    Partner getPartner(int id);

    @Delete
    void delete(Partner partner);

    @Query("DELETE FROM Partner")
    void deleteAll();
}

File: app/src/main/java/com/example/paetz/yacguide/database/PartnerDao.kt
package com.example.paetz.yacguide.database

import android.arch.persistence.room.Dao
import android.arch.persistence.room.Delete
import android.arch.persistence.room.Insert
import android.arch.persistence.room.OnConflictStrategy
import android.arch.persistence.room.Query

@Dao
interface PartnerDao {
    @get:Query("SELECT * FROM Partner")
    val all: Array<Partner>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(partner: Partner)

    @Query("SELECT Id FROM Partner WHERE Name = :name")
    fun getId(name: String): Int

    @Query("SELECT * FROM Partner WHERE Id = :id")
    fun getPartner(id: Int): Partner?

    @Delete
    fun delete(partner: Partner)

    @Query("DELETE FROM Partner")
    fun deleteAll()
}
